src/metal/rt64_metal.cpp

```diff
@@ -2632,32 +2672,48 @@ namespace RT64 {
         if (activeRenderEncoder == nullptr) {
             NS::AutoreleasePool *releasePool = NS::AutoreleasePool::alloc()->init();
 
-            // target frame buffer & sample positions affect the descriptor
             MTL::RenderPassDescriptor *renderDescriptor = MTL::RenderPassDescriptor::renderPassDescriptor();
 
+            // Setup color attachments
             for (uint32_t i = 0; i < targetFramebuffer->colorAttachments.size(); i++) {
                 MTL::RenderPassColorAttachmentDescriptor *colorAttachment = renderDescriptor->colorAttachments()->object(i);
                 colorAttachment->setTexture(targetFramebuffer->colorAttachments[i]->getTexture());
-                colorAttachment->setLoadAction(MTL::LoadActionLoad);
+                colorAttachment->setLoadAction(pendingClears.active ? pendingClears.initialAction[i] : MTL::LoadActionLoad);
+                if (pendingClears.active && pendingClears.initialAction[i] == MTL::LoadActionClear) {
+                    colorAttachment->setClearColor(mapClearColor(pendingClears.clearValues[i].color));
+                }
                 colorAttachment->setStoreAction(MTL::StoreActionStore);
             }
 
+            // Setup depth attachment
             if (targetFramebuffer->depthAttachment != nullptr) {
+                const size_t depthIndex = targetFramebuffer->colorAttachments.size();
                 MTL::RenderPassDepthAttachmentDescriptor *depthAttachment = renderDescriptor->depthAttachment();
                 depthAttachment->setTexture(targetFramebuffer->depthAttachment->mtl);
-                depthAttachment->setLoadAction(MTL::LoadActionLoad);
+                depthAttachment->setLoadAction(pendingClears.active ? pendingClears.initialAction[depthIndex] : MTL::LoadActionLoad);
+                if (pendingClears.active && pendingClears.initialAction[depthIndex] == MTL::LoadActionClear) {
+                    depthAttachment->setClearDepth(pendingClears.clearValues[depthIndex].depth);
+                }
```

Comment:
Same as above, skipping the check for active and for the clear action should make this code much simpler.